Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -139,6 +139,44 @@ theorem splits_X_pow (n : ℕ) : (X ^ n).Splits i :=
 theorem splits_id_iff_splits {f : K[X]} : (f.map i).Splits (RingHom.id L) ↔ f.Splits i := by
   rw [splits_map_iff, RingHom.id_comp]
 
+theorem Splits.comp_X_sub_C {i : L →+* F} (a : L) {f : L[X]}
+    (h : f.Splits i) : (f.comp (X - C a)).Splits i := by
+  cases h with
+  | inl h0 =>
+    left
+    simp only [map_eq_zero] at h0 ⊢
+    exact h0.symm ▸ zero_comp
```

Comment:
You don't need L be a field here, maybe you can move it to an appropriate section.
```suggestion
theorem Splits.comp_X_sub_C {F L} [CommRing L] [Field F] {i : L →+* F} (a : L) {f : L[X]}
    (h : f.Splits i) : (f.comp (X - C a)).Splits i := by
  cases h with
  | inl h0 =>
    left
    rw [map_comp, h0, zero_comp]
```
And this would better be stated as an iff. Even better, can you prove `(f.comp g).Splits i ↔ f.Splits i` if `degree (map i g) = 1`? This would require constructing an `algEquiv` encompassing `algEquivAevalXAddC` and `algEquivAevalNegX`, but would avoid the same argument being repeating in `Splits.comp_neg_X` below.

Reply:
Thank you for your suggestion! @alreadydone 

I will add the iff version. However, `(f.comp g).Splits i ↔ f.Splits i` if `degree (map i g) = 1` do need `L` to be a field. Do you suggest
1. I keep both versions, or 
2. only add an iff version for `CommRing`
3. only prove a more general form for `Field`

Reply:
I don't think it need L (I think you mean the domain of the RingHom `i`, which is in fact K after my change) to be a field. [Polynomial.splits_id_iff_splits](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Splits.html#Polynomial.splits_id_iff_splits) should have made it clear that splitting only depends on the image of the polynomial under the `i`, which always have coefficients in a field (`L`).

Reply:
Thank you! I've restated the theorems in `K`.